model/task/db.go

```diff
@@ -788,25 +788,36 @@ func needsContainerAllocation() bson.M {
 //     have been met.
 //  2. The container is allocated but the agent has not picked up the task yet.
 func IsContainerTaskScheduledQuery() bson.M {
+	query := IsUndispatchedContainerTasksQuery()
+	query["$or"] = dependenciesAreMetQuery()
+	return query
+}
+
+// IsUndispatchedContainerTasksQuery returns a query returning all tasks that
+// are undispatched container tasks.
+func IsUndispatchedContainerTasksQuery() bson.M {
 	return bson.M{
 		StatusKey:            evergreen.TaskUndispatched,
 		ActivatedKey:         true,
 		ExecutionPlatformKey: ExecutionPlatformContainer,
 		PriorityKey:          bson.M{"$gt": evergreen.DisabledTaskPriority},
-		"$or": []bson.M{
-			{
-				DependsOnKey: bson.M{"$size": 0},
-			},
-			{
-				// Containers can only be allocated for tasks whose dependencies
-				// are all met. All dependencies are met if they're all finished
-				// running and are still attainable (i.e. the dependency's
-				// required status matched the task's actual ending status).
-				bsonutil.GetDottedKeyName(DependsOnKey, DependencyFinishedKey):     true,
-				bsonutil.GetDottedKeyName(DependsOnKey, DependencyUnattainableKey): bson.M{"$ne": true},
-			},
-			{OverrideDependenciesKey: true},
+	}
+}
+
+func dependenciesAreMetQuery() []bson.M {
```

Comment:
Since this is only used in IsContainerTaskScheduledQuery, I think there's not a benefit to factoring it out; easier to read it all together

Reply:
This was more done to allow the `UndispatchedContainerTasksQuery` function to be reused in multiple places (one invocation needs this `dependenciesAreMetQuery ` appended to it, and the other one (part of the check blocked tasks job) needs to exclude it

Reply:
Right but since only one needs it, we can just define it there (i.e. in ScheduledContainerTasksQuery) explicitly (make sense to keep undispatchedContainerTasksQuery factored out since it's used in more places)